alacritty_terminal/src/ansi.rs

```diff
@@ -28,28 +28,64 @@ use crate::term::color::Rgb;
 //
 // Expect that color argument looks like "rgb:xx/xx/xx" or "#xxxxxx"
 fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
-    let mut iter = color.iter();
+    let mut iter = color.iter().peekable();
 
     macro_rules! next {
         () => {
             iter.next().map(|v| *v as char)
         };
     }
 
+    macro_rules! peek {
+        () => {
+            iter.peek().map(|v| **v as char)
+        };
+    }
+
+    macro_rules! convert_digit {
+        ($next:expr) => { $next.and_then(|v| v.to_digit(16)) };
+    }
+
     macro_rules! parse_hex {
         () => {{
-            let mut digit: u8 = 0;
-            let next = next!().and_then(|v| v.to_digit(16));
-            if let Some(value) = next {
-                digit = value as u8;
+            let mut value: u8 = 0;
+            if let Some(digit) = convert_digit!(next!()) {
+                value = digit as u8;
+                value <<= 4;
+            }
+            if let Some(digit) = convert_digit!(next!()) {
+                value += digit as u8;
+            }
+            value
+        }};
+    }
+
+    macro_rules! parse_separated_hex {
+        () => {{
+            let next = next!();
+            if next == Some('/') {
+                return None;
             }
 
-            let next = next!().and_then(|v| v.to_digit(16));
-            if let Some(value) = next {
-                digit <<= 4;
-                digit += value as u8;
+            let mut value: u8 = 0;
+            if let Some(digit) = convert_digit!(next) {
+                value = digit as u8;
+                value <<= 4;
             }
-            digit
+
+            if peek!() != Some('/') {
```

Comment:
I'm doing a last once-over and I realized that this structure will eat the bell character. Not eating the bell character was the entire reason for making the iterator peekable. Fixing this now.